File: scripts/keep_supabase_alive.py
import time
import requests

SUPABASE_URL = "https://knopoetvssfyxmvggqei.supabase.co"
SUPABASE_KEY = "sb_publishable_-lTSqONdT5KgK3D2d8102Q_8t9yXYbe"

headers = {
    "apikey": SUPABASE_KEY,
    "Authorization": f"Bearer {SUPABASE_KEY}"
}

print("🚀 Supabase Anti-Pause Keep-Alive Daemon Started...")

def ping():
    try:
        url = f"{SUPABASE_URL}/rest/v1/"
        response = requests.get(url, headers=headers, timeout=10)
        print(f"[{time.strftime('%Y-%m-%d %H:%M:%S')}] Pinging Supabase... Status Code: {response.status_code}")
    except Exception as e:
        print(f"[{time.strftime('%Y-%m-%d %H:%M:%S')}] Ping failed: {e}")

if __name__ == "__main__":
    while True:
        ping()
        # Sleep 6 hours
        time.sleep(21600)

File: scripts/keep_supabase_alive.php
<?php
// Supabase Anti-Pause Keep-Alive PHP Script
// Can be executed via cron: php keep_supabase_alive.php

$supabaseUrl = "https://knopoetvssfyxmvggqei.supabase.co";
$supabaseKey = "sb_publishable_-lTSqONdT5KgK3D2d8102Q_8t9yXYbe";

$ch = curl_init();
curl_setopt($ch, CURLOPT_URL, $supabaseUrl . "/rest/v1/");
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
curl_setopt($ch, CURLOPT_HTTPHEADER, [
    "apikey: " . $supabaseKey,
    "Authorization: Bearer " . $supabaseKey
]);
curl_setopt($ch, CURLOPT_TIMEOUT, 10);

$response = curl_exec($ch);
$httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
curl_close($ch);

echo "[" . date("Y-m-d H:i:s") . "] Supabase Pinged! HTTP Status Code: " . $httpCode . "\n";
?>
